digdag-standards/src/main/java/io/digdag/standards/operator/td/TdWaitOperatorFactory.java

```diff
@@ -134,9 +145,24 @@ private String startJob(TDOperator op, String domainKey)
 
         private boolean fetchJobResult(TDJobOperator job)
         {
-            // TODO: handle netsplits
+            Config resultState = state.getNestedOrSetEmpty(RESULT);
+
+            Optional<ArrayValue> firstRow;
+            try {
+                firstRow = job.getResult(ite -> ite.hasNext() ? Optional.of(ite.next()) : Optional.absent());
+            }
+            catch (UncheckedIOException | TDClientException e) {
+                if (isDeterministicClientException(e)) {
+                    throw new TaskExecutionException(e, TaskExecutionException.buildExceptionErrorConfig(e));
+                }
+                int retry = resultState.get(RETRY, int.class, 0);
+                resultState.set(RETRY, retry + 1);
+                int interval = (int) Math.min(INITIAL_RESULT_FETCH_RETRY_INTERVAL * Math.pow(2, retry), MAX_RESULT_FETCH_RETRY_INTERVAL);
+                logger.warn("Failed to download result of job '{}', retrying in {} seconds", job.getJobId(), interval, e);
+                throw TaskExecutionException.ofNextPolling(interval, ConfigElement.copyOf(state));
+            }
 
-            Optional<ArrayValue> firstRow = job.getResult(ite -> ite.hasNext() ? Optional.of(ite.next()) : Optional.absent());
+            state.remove(RESULT);
```

Comment:
[trivial] I'm not sure which is better, `resultState.remove(RETRY)` or `state.remove(RESULT)`...